File: web/src/api/contactList.js
import service from '@/utils/request'

export const getContactListTree = () => {
  return service({
    url: '/contactList/tree',
    method: 'get'
  })
}

export const createContactList = (data) => {
  return service({
    url: '/contactList/create',
    method: 'post',
    data
  })
}

export const updateContactList = (data) => {
  return service({
    url: '/contactList/update',
    method: 'put',
    data
  })
}

export const deleteContactList = (data) => {
  return service({
    url: '/contactList/delete',
    method: 'delete',
    data
  })
}

File: server/router/system/contact_list.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ContactListRouter struct{}

func (r *ContactListRouter) InitContactListRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	group := Router.Group("contactList").Use(middleware.OperationRecord())
	groupWithoutRecord := Router.Group("contactList")
	{
		group.POST("create", contactListApi.CreateContactList)
		group.PUT("update", contactListApi.UpdateContactList)
		group.DELETE("delete", contactListApi.DeleteContactList)
	}
	{
		groupWithoutRecord.GET("tree", contactListApi.GetContactListTree)
	}
}

